src/build/__main__.py

```diff
@@ -126,6 +119,56 @@ def build_package(srcdir, outdir, distributions, config_settings=None, isolation
         _error(str(e))
 
 
+def build_package(srcdir, outdir, distributions, config_settings=None, isolation=True, skip_dependency_check=False):
+    # type: (str, str, List[str], Optional[ConfigSettingsType], bool, bool) -> None
+    """
+    Run the build process.
+
+    :param srcdir: Source directory
+    :param outdir: Output directory
+    :param distribution: Distribution to build (sdist or wheel)
+    :param config_settings: Configuration settings to be passed to the backend
+    :param isolation: Isolate the build in a separate environment
+    :param skip_dependency_check: Do not perform the dependency check
+    """
+    with _handle_build_error():
+        builder = ProjectBuilder(srcdir)
+        for distribution in distributions:
+            _build(isolation, builder, outdir, distribution, config_settings, skip_dependency_check)
+
+
+def build_package_via_sdist(srcdir, outdir, distributions, config_settings=None, isolation=True, skip_dependency_check=False):
+    # type: (str, str, List[str], Optional[ConfigSettingsType], bool, bool) -> None
+    """
+    Build a sdist and then the specified distributions from it.
+
+    :param srcdir: Source directory
+    :param outdir: Output directory
+    :param distribution: Distribution to build (only wheel)
+    :param config_settings: Configuration settings to be passed to the backend
+    :param isolation: Isolate the build in a separate environment
+    :param skip_dependency_check: Do not perform the dependency check
+    """
+    if 'sdist' in distributions:
+        raise ValueError('Only binary distributions are allowed but sdist was specified')
```

Comment:
Looks like this is only done for simplicity in `main`. Suggest renaming the function to `build_sdist_and_wheel_from_sdist` and removing the `distributions` parameter. In `main` you can use a `partial`.

Reply:
No, it was a deliberate design decision to keep the API future-proof.

Reply:
Okay, I didn't really think of anything in `__main__` as public API.